test/sql/function/test_stringfunctions.cpp

```diff
@@ -463,6 +463,73 @@ TEST_CASE("LTRIM/RTRIM test", "[function]") {
 	REQUIRE_FAIL(con.Query("select RTRIM('hello', 'world')"));
 }
 
+TEST_CASE("LEFT test", "[function]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+	con.EnableQueryVerification();
+
+	// test LEFT on positive positions
+	result = con.Query("SELECT LEFT('abcd', 0), LEFT('abc', 1), LEFT('abc', 2), LEFT('abc', 3), LEFT('abc', 4)");
+	REQUIRE(CHECK_COLUMN(result, 0, {""}));
+	REQUIRE(CHECK_COLUMN(result, 1, {"a"}));
+	REQUIRE(CHECK_COLUMN(result, 2, {"ab"}));
+	REQUIRE(CHECK_COLUMN(result, 3, {"abc"}));
+	REQUIRE(CHECK_COLUMN(result, 4, {"abc"}));
+
+	result = con.Query("SELECT LEFT('🦆ab', 0), LEFT('🦆ab', 1), LEFT('🦆ab', 2), LEFT('🦆ab', 3), LEFT('🦆ab', 4)");
+	REQUIRE(CHECK_COLUMN(result, 0, {""}));
+	REQUIRE(CHECK_COLUMN(result, 1, {"🦆"}));
+	REQUIRE(CHECK_COLUMN(result, 2, {"🦆a"}));
+	REQUIRE(CHECK_COLUMN(result, 3, {"🦆ab"}));
+	REQUIRE(CHECK_COLUMN(result, 4, {"🦆ab"}));
+
+	// test LEFT on negative positions
+	result = con.Query("SELECT LEFT('abcd', 0), LEFT('abc', -1), LEFT('abc', -2), LEFT('abc', -3), LEFT('abc', -4)");
+	REQUIRE(CHECK_COLUMN(result, 0, {""}));
+	REQUIRE(CHECK_COLUMN(result, 1, {"ab"}));
+	REQUIRE(CHECK_COLUMN(result, 2, {"a"}));
+	REQUIRE(CHECK_COLUMN(result, 3, {""}));
+	REQUIRE(CHECK_COLUMN(result, 4, {""}));
+
+	result = con.Query("SELECT LEFT('🦆ab', 0), LEFT('🦆ab', -1), LEFT('🦆ab', -2), LEFT('🦆ab', -3), LEFT('🦆ab', -4)");
+	REQUIRE(CHECK_COLUMN(result, 0, {""}));
+	REQUIRE(CHECK_COLUMN(result, 1, {"🦆a"}));
+	REQUIRE(CHECK_COLUMN(result, 2, {"🦆"}));
+	REQUIRE(CHECK_COLUMN(result, 3, {""}));
+	REQUIRE(CHECK_COLUMN(result, 4, {""}));
+
+	// test LEFT on NULL values
+	result = con.Query("SELECT LEFT(NULL, 0), LEFT('abc', NULL), LEFT(NULL, NULL)");
+	REQUIRE(CHECK_COLUMN(result, 0, {""}));
+	REQUIRE(CHECK_COLUMN(result, 1, {""}));
+	REQUIRE(CHECK_COLUMN(result, 2, {""}));
+
+	result = con.Query("SELECT LEFT(NULL, 0), LEFT('🦆ab', NULL), LEFT(NULL, NULL)");
```

Comment:
Could you also add a test that runs this on the 🤦🏼‍♂️ and S̈ characters? These have combining unicode glyphs.

Reply:
Done, PTAL